src/main/java/seedu/address/commons/util/CollectionUtil.java

```diff
@@ -32,4 +35,34 @@ public static void requireAllNonNull(Collection<?> items) {
     public static boolean isAnyNonNull(Object... items) {
         return items != null && Arrays.stream(items).anyMatch(Objects::nonNull);
     }
+
+    /**
+     * Zips two streams together with a BiFunction.
+     * Solution adapted from
+     * "<a href="https://stackoverflow.com/questions
+     * /17640754/zipping-streams-using-jdk8-with-lambda
+     * -java-util-stream-streams-zip">Stack Overflow</a>".
+     *
+     * Apparently, there's no raw Java that can zip streams together.
+     * If you can find alternatives, you can just take this out.
```

Comment:
LMK if there's any alternative to implementing our own zip method.

Reply:
LGTM